UnityWeld/Binding/AbstractMemberBinding.cs

```diff
@@ -40,47 +39,50 @@ public virtual void Init()
-                    .FirstOrDefault(component => component.GetType().ToString() == viewModelName);
-                if (monoBehaviourViewModel != null)
+                using(var cache = trans.gameObject.GetComponentsWithCache<MonoBehaviour>(false))
                 {
-                    return monoBehaviourViewModel;
-                }
-
-                var providedViewModel = buffer
-                    .Select(component => component.GetViewModelData())
-                    .Where(component => component != null)
-                    .FirstOrDefault(
-                        viewModelData => viewModelData.TypeName == viewModelName &&
+                    var monoBehaviourViewModel = cache.Components
+                                                      .FirstOrDefault(component =>
+                                                                          component.GetType().ToString() ==
+                                                                          viewModelName);
+                    if(monoBehaviourViewModel != null)
+                    {
+                        return monoBehaviourViewModel;
+                    }
+
+                    var providedViewModel = cache.Components
+                                                 .Select(component => component.GetViewModelData())
+                                                 .Where(component => component != null)
+                                                 .FirstOrDefault(
+                                                     viewModelData => viewModelData.TypeName == viewModelName &&
 #pragma warning disable 252,253 // Warning says unintended reference comparison, but we do want to compare references
-                                         (object) viewModelData.Model != this
+                                                                      (object)viewModelData.Model != this
 #pragma warning restore 252,253
-                    );
+                                                 );
 
-                if (providedViewModel != null)
-                {
-                    return providedViewModel.Model;
+                    if(providedViewModel != null)
+                    {
+                        return providedViewModel.Model;
+                    }
                 }
 
                 trans = trans.parent;
             }
 
-            throw new ViewModelNotFoundException(string.Format(
-                "Tried to get view model {0} but it could not be found on "
-                + "object {1}. Check that a ViewModelBinding for that view model exists further up in "
-                + "the scene hierarchy. ", viewModelName, gameObject.name)
+            throw new ViewModelNotFoundException(
+                $"Tried to get view model {viewModelName} but it could not be found on " +
+                $"object {gameObject.name}. Check that a ViewModelBinding for that view model exists further up in " +
+                "the scene hierarchy. "

```

Comment:
I'd prefer not to change from `string.Format` to string interpolation since it will break the code in older versions of Unity, although I guess if this PR still requires .NET 4.x support then I think all versions of Unity that support that should work with this as well. In general I'd prefer not to change stuff like this unless necessary though.